src/api/nodeRpcCall.js

```diff
@@ -1925,6 +2029,191 @@ const fetchPendingIdentities = async () => {
   return processed.filter((entity) => entity.data.judgements.length === 0);
 };
 
+const getSwapPriceExactTokensForTokens = async (asset1, asset2, amount, includeTax = true) => {
+  const api = await getApi();
+
+  const maybeRate = await api.call.assetConversionApi.quotePriceExactTokensForTokens(
+    asset1,
+    asset2,
+    amount,
+    includeTax,
+  );
+  if (maybeRate.isNone) {
+    return null;
+  }
+  return maybeRate.unwrap();
+};
+
+const getSwapPriceTokensForExactTokens = async (asset1, asset2, amount, includeTax = true) => {
+  const api = await getApi();
+  const maybeRate = await api.call.assetConversionApi.quotePriceTokensForExactTokens(
+    asset1,
+    asset2,
+    amount,
+    includeTax,
+  );
+  if (maybeRate.isNone) {
+    return null;
+  }
+  return maybeRate.unwrap();
+};
+
+const getDexReserves = async (asset1, asset2) => {
+  const api = await getApi();
+  const maybeReserves = await api.call.assetConversionApi.getReserves(asset1, asset2);
+  if (maybeReserves.isNone) {
+    return null;
+  }
+  const [reservesOfAsset1, reservesOfAsset2] = maybeReserves.unwrap();
+  return {
+    asset1: reservesOfAsset1,
+    asset2: reservesOfAsset2,
+  };
+};
+
+const getLpTokensOwnedByAddress = async (lpTokenId, address) => {
+  const api = await getApi();
+  const maybeTokens = await api.query.poolAssets.account(lpTokenId, address);
+
+  if (maybeTokens.isNone) {
+    return null;
+  }
+  const tokens = maybeTokens.unwrapOrDefault();
+  const balance = tokens.balance.toString();
+  return { balance };
+};
+
+const getDexPools = async (walletAddress) => {
+  try {
+    const api = await getApi();
+    const pools = await api.query.assetConversion.pools.entries();
+    const poolsData = await Promise.all(pools.map(async ([poolKey, maybePoolData]) => {
+      const [asset1, asset2] = poolKey.args[0];
+      const { lpToken } = maybePoolData.unwrapOrDefault();
+      const asset1checkIsNative = asset1.value.toString();
+      const asset2checkIsNative = asset2.value.toString();
+      const asset2Tronsform = asset2checkIsNative || asset2.toString();
```

Comment:
```suggestion
      const asset2Transform = asset2checkIsNative || asset2.toString();
```
This typo shows up in other places too, please fix all.